### PR TITLE
Doc 472/custom pack registry updates

### PR DESCRIPTION
**Pull request by oz123**
## Describe the Change

This PR extends the documentation of the PAX registry. It adds information on previously
undocumented features and shows how to deploy the registry with letsencrypt, which was added in v3.2.

## Review Changes

🎫 [Jira ticket](https://spectrocloud.atlassian.net/browse/DOC-295)

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1217#" title="Author self-approved">oz123</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by spectro-prow** [comment]
Hi @oz123. Thanks for your PR.

I'm waiting for a [spectrocloud](https://github.com/orgs/spectrocloud/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/spectrocloud/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (oz123) [reply content/docs/09-registries-and-packs/7-advanced-configuration.md:190]: yes.

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (oz123) [reply content/docs/09-registries-and-packs/7-advanced-configuration.md:191]: yes.

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (oz123) [reply content/docs/09-registries-and-packs/7-advanced-configuration.md:195]: yes.

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (oz123) [reply content/docs/09-registries-and-packs/7-advanced-configuration.md:210]: yes.

Comment (oz123) [reply content/docs/09-registries-and-packs/7-advanced-configuration.md:210]: yes.

File: content/docs/09-registries-and-packs/1-adding-a-custom-registry.md
Comment (oz123) [reply content/docs/09-registries-and-packs/1-adding-a-custom-registry.md:39]: yes.

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (oz123) [reply content/docs/09-registries-and-packs/7-advanced-configuration.md:206]: yes.

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (oz123) [reply content/docs/09-registries-and-packs/7-advanced-configuration.md:189]: yes.

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (oz123) [reply content/docs/09-registries-and-packs/7-advanced-configuration.md:169]: yes.

File: content/docs/09-registries-and-packs/1-adding-a-custom-registry.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/1-adding-a-custom-registry.md:39]:
```suggestion
# Deploy a Pack Registry Server with SSL Certificates
```
Just trying to think of ways to shorten the title.

File: content/docs/09-registries-and-packs/1-adding-a-custom-registry.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/1-adding-a-custom-registry.md:42]:
```suggestion

Check out the [advanced configuration](/registries-and-packs/adding-a-custom-registry#self-signedcertificates) section for more advanced configuration and deployment with self-signed certificates.
```
Added a newline and changed the order to improve the flow.

File: content/docs/09-registries-and-packs/1-adding-a-custom-registry.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/1-adding-a-custom-registry.md:47]:
```suggestion
1. Create a folder that contains an httppasswd file. 
```

File: content/docs/09-registries-and-packs/1-adding-a-custom-registry.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/1-adding-a-custom-registry.md:50]: This step reads better if split it out. Here is a previous example. 

CleanShot 2023-03-29 at 16 21 49

File: content/docs/09-registries-and-packs/1-adding-a-custom-registry.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/1-adding-a-custom-registry.md:55]:
```suggestion
2. Create a pax registry configuration file titled **myconfig.yml**. This file should be in the **spectropaxconfig** directory.
```

File: content/docs/09-registries-and-packs/1-adding-a-custom-registry.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/1-adding-a-custom-registry.md:76]:
```suggestion
3. Start the container image with the following flags.
```

File: content/docs/09-registries-and-packs/1-adding-a-custom-registry.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/1-adding-a-custom-registry.md:89]:
```suggestion
You can now access the pack registry at `https://yourhost.companydomain.com/v1/`.
You will be prompted to give the user admin and the password of your choice.
```

File: content/docs/09-registries-and-packs/1-adding-a-custom-registry.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/1-adding-a-custom-registry.md:91]:
```suggestion
# Deploy a Pack Registry Server with Self-Signed Certificates
```
We avoid starting titles with verbs that end in -ing.

File: content/docs/09-registries-and-packs/1-adding-a-custom-registry.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/1-adding-a-custom-registry.md:93]:
```suggestion
The following steps need to be performed to deploy the pack registry server using self-signed certificates:
```

File: content/docs/09-registries-and-packs/1-adding-a-custom-registry.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/1-adding-a-custom-registry.md:129]:
```suggestion
 docker pull gcr.io/spectro-images-public/release/spectro-registry:3.3.0
```

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:2]:
```suggestion
title: 'Advanced Configuration'
```
A bit shorter and to the point. We are already in the context of registries, so it might be a bit redundant.

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:3]:
```suggestion
metaTitle: 'Advanced Configuration'
```

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:4]:
```suggestion
metaDescription: 'Learn how to apply advanced concepts by customizing the deployments of the Packs registry.'
```

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:17]:
```suggestion
# Advanced Configuration
```

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:24]:
```suggestion
You can modify a few deployment options of the pack registry.
The configuration is applied through a YAML file. However, you can override 
many options using environment variables.

The configuration file is divided into keys and values. The following is an example YAML configuration.
```

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:26]:
```suggestion
```yaml
```

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:36]:
```suggestion
The key `version` has a number value. The `log` key has a value with multiple keys, which in turn have more keys.
```

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:39]:
```suggestion
`REGISTRY_LOG_LEVEL`.

```shell
export REGISTRY_LOG_LEVEL=debug
```

```

Might be good to provide an example.

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:41]:
```suggestion
## Default Configuration
```

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:43]:
```suggestion
The docker image for the registry contains the following default configuration values.

<br />
```

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:72]:
```suggestion
You can override the default values with specific values through environment
variables, or you can use your own configuration file.
```

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:75]:
```suggestion
For example, you can start the docker container image with the following environment
variables to override the basic auth realm and logging level:
```

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:89]:
```suggestion
Alternatively, you can start the container by mounting a directory with a new configuration file and pointing the server command to the configuration file.
```

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:97]:
```suggestion
    gcr.io/spectro-images-public/release/spectro-registry:3.3.0
```
Should this be 3.3.0?

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:103]:
```suggestion
The pack registry can store data on a file system through a mounted
volume, or you can specify object storage such as AWS S3.
```

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:105]:
```suggestion
The following is an example of a configuration using a file system backend.
```

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:107]:
```suggestion
```yaml
```

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:115]:
```suggestion
If you are using S3 Storage, ensure you specify the required S3 parameters.
```

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:133]:
```suggestion
You can also use ephemeral storage. We recommend using ephemeral storage for testing purposes. Production environments should use object storage or a file system.
```

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:134]:
```suggestion
```yaml
```

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:140]:
```suggestion
You can configure basic HTTP Auth. Basic Auth requires providing the pack registry server with an httppasswd file containing the credentials.
```

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:142]:
```suggestion
```yaml
```

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:149]:
```suggestion
## Network
```
I wonder if network is a better topic name.

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:161]:
```suggestion
Alternatively, you can bind to a single IP and different port:
```
Adress the user :)

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:178]:
```suggestion
    Referrer-Policy: [ same-origin ]
```
```

Missing closing ticks

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (karl-cardenas-coding) [inline content/docs/09-registries-and-packs/7-advanced-configuration.md:185]:
```suggestion
validation by the letsencrypt services.
```

**Review by karl-cardenas-coding — CHANGES_REQUESTED** [review]
@oz123 thank you so much for this PR. I left you some suggestions that can help the reading experience. There are some markdown issues on the advanced configuration page. Ping me once you address it, and we can get this to production 🚀

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (oz123) [reply content/docs/09-registries-and-packs/7-advanced-configuration.md:149]: I tried keeping the sections with the same name as in the YAML configuration file.

**Comment by oz123** [comment]
@karl-cardenas-coding thanks for the thorough review!

File: content/docs/09-registries-and-packs/1-adding-a-custom-registry.md
Comment (oz123) [reply content/docs/09-registries-and-packs/1-adding-a-custom-registry.md]: Fixed.

File: content/docs/09-registries-and-packs/1-adding-a-custom-registry.md
Comment (oz123) [reply content/docs/09-registries-and-packs/1-adding-a-custom-registry.md]: Fixed.

File: content/docs/09-registries-and-packs/1-adding-a-custom-registry.md
Comment (oz123) [reply content/docs/09-registries-and-packs/1-adding-a-custom-registry.md]: Adapted a bit. Hope the new title is better.

File: content/docs/09-registries-and-packs/7-advanced-configuration.md
Comment (oz123) [reply content/docs/09-registries-and-packs/7-advanced-configuration.md:149]: We can change to Network but, that title covers address and binding to ports too. Hence, I feel HTTP is more accurate.

Comment (karl-cardenas-coding) [reply content/docs/09-registries-and-packs/7-advanced-configuration.md:149]: I follow now. In that case, let's keep it as is.

File: content/docs/09-registries-and-packs/1-adding-a-custom-registry.md
Comment (oz123) [reply content/docs/09-registries-and-packs/1-adding-a-custom-registry.md:135]: nope. not here ...

**Review by karl-cardenas-coding — APPROVED** [review]
Awesome PR, thank @oz123

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/1217#pullrequestreview-1365825775" title="Approved">karl-cardenas-coding</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/1217#" title="Author self-approved">oz123</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
